dummyserver/hypercornserver.py

```diff
@@ -62,7 +76,7 @@ def run_hypercorn_in_thread(
         if not ready_event.is_set():
             raise Exception("most likely failed to start server")
 
-        yield
+        yield typing.cast(int, parse_url(config.bind[0]).port)
```

Comment:
```suggestion
        port = parse_url(config.bind[0]).port
        assert port is not None
        yield port
```

Reply:
When can this happen?

Reply:
OK I understand now that this is to avoid the cast.

Reply:
Done in [`54f28c9` (#3234)](https://github.com/urllib3/urllib3/pull/3234/commits/54f28c9b38d10c656a35d9093dc6d15fee9a95f4)